packages/nextjs/components/viem/CloseBets.tsx

```diff
@@ -1,17 +1,44 @@
-import { useCallback } from "react";
+import { useCallback, useContext, useState } from "react";
+import TransactionList from "./TransactionList";
+import { useContractWrite } from "wagmi";
+import * as chains from "wagmi/chains";
+import ErrorBlock from "~~/components/Error";
+import { ContractContext } from "~~/context";
+import deployedContracts from "~~/contracts/deployedContracts";
 
 export default function CloseBets({ className }: { className?: string }) {
-  const onClick = useCallback(async () => {
-    console.log("Close bets");
-  }, []);
+  const [txHashes, setTxHashes] = useState<string[]>([]);
+
+  const contractContext = useContext(ContractContext);
+
+  const { write, error } = useContractWrite({
+    address: contractContext.lotteryAddress,
+    abi: deployedContracts[chains.sepolia.id].Lottery.abi,
+    functionName: "closeLottery",
+    onSuccess: useCallback(
+      (data: { hash: string }) => {
+        setTxHashes(prev => [data.hash, ...prev]);
+      },
+      [setTxHashes],
+    ),
+  });
+
+  const onSubmit = useCallback(() => {
+    write({
+      args: [],
+    });
```

Comment:
```suggestion
    write();
```
You can just omit parameters